### ISSUE TITLE
Cl() is not properly documented

### ISSUE THREAD

**Issue by arsenovic** [pygae/clifford#379]
https://github.com/pygae/clifford/blob/master/clifford/__init__.py#L352

this is possibly the most used function in this package.

**Comment by eric-wieser** [comment]
In what sense do you mean "properly"?

**Comment by arsenovic** [comment]
dont most functions have their parameters listed under 

    Parameters
    -----------------

**Comment by eric-wieser** [comment]
That's true, but I think describing `p, q, r` in the main text works fine anyway. The function is messy to document because it supports multiple unrelated calling conventions...

We should probably at least add an example of each one to the docstring.

https://clifford.readthedocs.io/en/latest/api/clifford.Cl.html#clifford.Cl

**Comment by hugohadfield** [comment]
Yeah I think it definitely needs a bit of a flesh out documentation-wise
